osf/models/node.py

```diff
@@ -2235,6 +2235,40 @@ def storage_usage(self):
             update_storage_usage(self)  # sets cache
             return storage_usage_cache.get(key)
 
+    def add_contributor(self, *args, **kwargs):
+        contributor = super(AbstractNode, self).add_contributor(*args, **kwargs)
+        if contributor and not contributor.is_registered:
+            prereg_system_tag = self.all_tags.filter(name='source:campaign|prereg', system=True).first()
+            registered_report_system_tag = self.all_tags.filter(name='source:campaign|osf_registered_reports', system=True).first()
+            osf4m_system_tag = self.all_tags.filter(name='source:campaign|osf4m', system=True).first()
+            if prereg_system_tag:
+                contributor.add_system_tag(prereg_system_tag)
+            elif registered_report_system_tag:
+                contributor.add_system_tag(registered_report_system_tag)
+            elif osf4m_system_tag:
+                contributor.add_system_tag(osf4m_system_tag)
+            else:
+                osf_provider_tag, created = Tag.all_tags.get_or_create(name='source:provider|osf', system=True)
+                contributor.add_system_tag(osf_provider_tag)
+        return contributor
+
+    def add_unregistered_contributor(self, *args, **kwarg):
```

Comment:
This override the `add_unregistered_contributor` method defined in `ContributorMixin`. However this method is not currently used for the `Node` model. To add unregistered contributor for a node, the only way is posting to a V1 endpoint, whose view triggers the `add_contributors` method, which in turn triggers the `add_contributor` method defined on `ContributorMixin`. Therefor, `add_contributor` is also overridden to add corresponding tags to unregistered contributors.